### ISSUE TITLE
Possible memory leak at steady state

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#5759]
Reported on Origin 1.0.7 with multi-tenant SDN - no data as yet on the source of the leak.

**Comment by smarterclayton** [comment]
@ironcladlou FYI this is the tracking issue for the unconfirmed memory leak on a 1.0.7 cluster at steady state.  Only variable we know is that multi-tenant was on (no evidence it was the cause).

**Comment by smarterclayton** [comment]
Multi-tenant disabled had no effect on the leak.

**Comment by derekwaynecarr** [comment]
Is there any information on what the system was doing?  Is it just steady state no content?

**Comment by smarterclayton** [comment]
It had a fair amount of content.  I think it's dan maces deployment config
error thing only, but need to be sure.

On Nov 6, 2015, at 2:17 PM, Derek Carr <notifications@github.com> wrote:

Is there any information on what the system was doing? Is it just steady
state no content?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/issues/5759#issuecomment-154505437>.

**Comment by gravis** [comment]
We have the issue here, please let us what kind of info we can provide.
We just notice the process using more and more memory, even on a fresh install. For example, on the current test cluster, os is using 600Mo after 5 hours. If we put our projects on this cluster, openshift can eat 8GB RAM + all the swap in less than 2 minutes.
We have tried several configurations:

- With embedded etcd
- With a local etcd
- With a dedicated etcd server
- Without or without multi-tenant

In the best case, OS is just eating memory, in some we couldn't deploy anything.

**Comment by ironcladlou** [comment]
One way we could confirm it's related to #5504 would be to inspect the `status.details.message` field of `deploymentConfig` objects in etcd. If those fields are growing beyond a few hundred bytes, it's #5504.

**Comment by smarterclayton** [comment]
That's very likely it (was reported by Philippe), this is to track after we
cut 1.0.8.

On Nov 6, 2015, at 5:08 PM, Dan Mace <notifications@github.com> wrote:

One way we could confirm it's related to #5504
<https://github.com/openshift/origin/issues/5504> would be to inspect the
status.details.message field of deploymentConfig objects in etcd. If those
fields are growing beyond a few hundred bytes, it's #5504
<https://github.com/openshift/origin/issues/5504>.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/issues/5759#issuecomment-154557123>.

**Comment by gravis** [comment]
It doesn't seem, on the only project being tested on this cluster:

```
status:
  details:
    message: "The image trigger for image stream tag \"test-site:latest\" will
      have no effect because image stream tag \"test-site:latest\" does not
      exist.\n\tIf image stream tag \"test-site:latest\" is expected, check
      build config \"tech-angels-site\" which produces image stream tag \"test-site:latest\"."
```

the IS exists, but no image is being created by the deployment sti script, we have an expected error:

    F1106 14:54:25.350055       1 builder.go:55] Build error: dial tcp: i/o timeout

The same project builds fine on 1.0.6, with the same configuration.

**Comment by derekwaynecarr** [comment]
@gravis - can you open a new issue specific to the build config / image stream config issue?

**Comment by gravis** [comment]
We're waiting for the new rpms (1.0.8) to test if it's present. If not, I think we can close this directly (?)

**Comment by derekwaynecarr** [comment]
@gravis - ok, thanks

**Comment by ironcladlou** [comment]
@gravis 
>We're waiting for the new rpms (1.0.8) to test if it's present. If not, I think we can close this directly (?)

Where will you be getting the updated RPMs? Just trying to determine who or what you're waiting on in case I can assist.

**Comment by pweil-** [comment]
any update here?

**Comment by gravis** [comment]
I confirm the 1.0.8 is not eating memory like the 1.0.7.
This morning we using 750MB of memory on our master (was 580MB yesterday).
So I would say, the new timeouts solved the issue temporary. Yet, it seems to still leak slowly (at an acceptable rate for now).
Please let me know if we can provide more info.

**Comment by pweil-** [comment]
Thanks @gravis.  We will continue to look into it and open specific issues for anything new.

**Comment by pweil-** [comment]
(hit the wrong button.  Leaving this open since there is still a leak)

**Comment by smarterclayton** [comment]
A heap dump would be the best way to diagnose and triage this.  The
instructions here can guide you to doing that:
https://github.com/openshift/origin/blob/master/HACKING.md#performance-debugging

Using the web mode and capturing two heaps spaced out in time will probably
allow is to very quickly identify your issue.

On Nov 11, 2015, at 7:54 AM, Philippe Lafoucrière <notifications@github.com>
wrote:

I confirm the 1.0.8 is not eating memory like the 1.0.7.
This morning we using 750MB of memory on our master (was 580MB yesterday).
So I would say, the new timeouts solved the issue temporary. Yet, it seems
to still leak slowly (at an acceptable rate for now).
Please let me know if we can provide more info.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/issues/5759#issuecomment-155824192>.

**Comment by gravis** [comment]
805MB this morning, I have opened a ticket to get to heap dumps.

**Comment by ironcladlou** [comment]
@gravis any update here?

**Comment by gravis** [comment]
Our masters (1.0.8) are stable around 750MB. 
We're still wanting to find time to get a heap dump, but with our current load, it won't be before a while I think :(
Last week we tried a update the cert of the cluster to replace the self-signed one, and everything felt apart. Apparently, this prevented the master and node from accessing the etcd servers, and the master/nodes started to eat memory like crazy again.

**Comment by gravis** [comment]
1.1.0 has been very stable regarding memory usage, I think we can leave this issue behind us now.
I suspect the issue was related to the glusterfs endpoints disappearing, and retry timeouts too shorts.
Let's close this and move on.
Thanks
